whisperv6/peer.go

```diff
@@ -192,21 +201,35 @@ func (peer *Peer) expire() {
 	}
 }
 
+func (peer *Peer) reduceBundle(bundle []*Envelope) []*Envelope {
+	if peer.host.ratelimiter == nil {
+		return bundle
+	}
+	for i := range bundle {
+		size := int64(bundle[i].size())
+		if peer.host.ratelimiter.Egress.Available(peer.peer) < size {
+			return bundle[:i]
+		}
+		peer.host.ratelimiter.Egress.TakeAvailable(peer.peer, size)
+	}
+	return bundle
+}
+
 // broadcast iterates over the collection of envelopes and transmits yet unknown
 // ones over the network.
 func (peer *Peer) broadcast() error {
 	if peer.peer.IsFlaky() {
 		log.Trace("Waiting for a peer to restore communication", "ID", peer.peer.ID())
 		return nil
 	}
-	envelopes := peer.host.Envelopes()
+	envelopes := peer.host.Envelopes() // envelopes are read from hash map, so access is already randomized
```

Comment:
Sure about it? Is each read truly randomized?

Reply:
it is definitely not truly randomized, in a sense that distribution is not uniformal. but every iteration starts from different index.
tbh i don't see it as very important thing. i tend to think that fifo order might be marginally better in case of spam. but if we are comparing  truly random vs somewhat random it is hard to justify preference towards one